File: app/src/main/java/com/yibao/music/adapter/SongViewPagerAdapter.java
package com.yibao.music.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.yibao.music.fragment.SongCategoryFragment;

/**
 * 作者：Stran on 2017/3/23 03:31
 * 描述：${TODO}
 * 邮箱：strangermy@outlook.com
 *
 * @author Stran
 */
public class SongViewPagerAdapter
        extends FragmentStateAdapter {


    public SongViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return SongCategoryFragment.newInstance(position);
    }

    @Override
    public int getItemCount() {
        return 4;
    }

}

File: app/src/main/java/com/yibao/music/adapter/SongViewPagerAdapter.kt
package com.yibao.music.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.yibao.music.fragment.SongCategoryFragment
import com.yibao.music.viewmodel.SongViewModel

/**
 * 作者：Stran on 2017/3/23 03:31
 * 描述：${TODO}
 * 邮箱：strangermy@outlook.com
 *
 * @author Stran
 */
class SongViewPagerAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return SongCategoryFragment.newInstance(position)
    }

    override fun getItemCount(): Int {
        return 4
    }
}